derive-macros/src/lib.rs

```diff
@@ -136,6 +136,79 @@ fn gen_schema_fields(data: &Data) -> TokenStream {
     quote! { #(#schema_fields),* }
 }
 
+/// Derive an IntoEngineData trait for a struct that has all fields implement `Into<Scalar>`.
+///
+/// This is a relatively simple macro to produce the boilerplate for converting a struct into
+/// EngineData using the `create_one` method.
+///
+/// # Example
+/// ```ignore
+/// #[derive(Schema, IntoEngineData)]
+/// struct MyStruct {
+///    a: i32,
+///    b: String,
+/// }
+///
+/// let my_struct = MyStruct { a: 42, b: "Hello".to_string() };
+/// // typically used with ToSchema
+/// let schema = Arc::new(MyStruct::to_schema());
+/// // single-row EngineData
+/// let engine_data = my_struct.into_engine_data(schema, engine);
+/// ```
+#[proc_macro_derive(IntoEngineData)]
+pub fn into_engine_data_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let struct_name = &input.ident;
+    let fields = match &input.data {
+        Data::Struct(data) => match &data.fields {
+            Fields::Named(fields) => &fields.named,
+            _ => {
+                return Error::new(
+                    struct_name.span(),
+                    "IntoEngineData can only be derived for structs with named fields",
+                )
+                .to_compile_error()
+                .into()
+            }
+        },
+        _ => {
+            return Error::new(
+                struct_name.span(),
+                "IntoEngineData can only be derived for structs",
+            )
+            .to_compile_error()
+            .into()
+        }
+    };
```

Comment:
```suggestion
    let Data::Struct(Fields::Named(fields)) = &data.fields else {
      return Error::new(
        struct_name.span(),
        "IntoEngineData can only be derived for structs with named fields",
      ).to_compile_error().into()
    }
```

(untested, but I think this should work)

Reply:
ah nice thanks